src/main/java/dev/ai4j/openai4j/image/GenerateImagesResponse.java

```diff
@@ -0,0 +1,122 @@
+package dev.ai4j.openai4j.image;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents the response from the OpenAI API when generating images.
+ *
+ * @param data A list of image data, each containing an image URL, Base64 encoded JSON response, and revised prompt
+ */
+public class GenerateImagesResponse {
+
+  private final List<ImageData> data;
+
+  public GenerateImagesResponse(Builder builder) {
+    this.data = builder.data;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public List<ImageData> data() {
+    return data;
+  }
+
+  public static class ImageData {
+
+    private String url;
+    private final String b64Json;
+    private final String revisedPrompt;
+
+    public ImageData(String url, String b64Json, String revisedPrompt) {
+      this.url = url;
+      this.b64Json = b64Json;
+      this.revisedPrompt = revisedPrompt;
+    }
+
+    public String url() {
+      return url;
+    }
+
+    public String b64Json() {
+      return b64Json;
+    }
+
+    public String revisedPrompt() {
+      return revisedPrompt;
+    }
+
+    public void url(String url) {
```

Comment:
I am not sure it is right to override the original URL, this is quite an unobvious hebahior.
Also, if user wants to get both URL and a file, currently there is no way to get the original URL as it is overriden by a file path (which is also not exactly a URL)

Reply:
The intention was to keep the process simple as possible, actually nobody the original URL if the image will downloaded, since this URL is ephemeral (will be kept by OpenAI for only some hours/minutes). What could be a simple alternative?

Reply:
ok, makes sense!

Reply:
but probably makes sense to leave a small comment in `withPersisting`/`persistTo` javadoc and mention that `ImageData.url()` contains a path of the file that has been persisted instead of original URL